.jenkins/profiling.groovy

```diff
@@ -0,0 +1,100 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+// This shared library is available at https://github.com/ROCmSoftwarePlatform/rocJENKINS/
+@Library('rocJenkins@pong') _
+
+// This is file for internal AMD use.
+// If you are interested in running your own Jenkins, please raise a github issue for assistance.
+
+import com.amd.project.*
+import com.amd.docker.*
+import java.nio.file.Path
+
+def runCompileCommand(platform, project, jobName, boolean debug=false) {
+    project.paths.construct_build_prefix()
+
+    command = """#!/usr/bin/env bash
+            set -ex
+            hostname
+            cd ${project.paths.project_build_prefix}
+
+            export HOME=/home/jenkins
+
+            gfx_name="gfx90a"
+            arch_name="aldebaran"
+            logic_path="library/src/blas3/Tensile/Logic/asm_full"
+            repo_name="rocBLAS"
+
+            git clone --depth=1 https://github.com/ROCm/\$repo_name.git ../\$repo_name
+
+            TENSILE_PROFILE=ON Tensile/bin/TensileCreateLibrary \
+              \$PWD/../\$repo_name/\$logic_path/\$arch_name \
+              _build \
+              HIP \
+              --merge-files \
+              --separate-architecture \
+              --lazy-library-loading \
+              --no-short-file-names \
+              --no-library-print-debug \
+              --code-object-version=default \
+              --cxx-compiler=hipcc \
+              --jobs=32 \
+              --library-format=msgpack \
+              --architecture=\$gfx_name
+
+            find . -name "*.prof" -print 2>/dev/null
+            """
+
+    platform.runCommand(this, command)
+
+    archiveArtifacts artifacts: '**/*.prof'
+}
+
+def runCI(nodeDetails, jobName) {
+    def prj = new rocProject('Tensile', 'Profiling')
+
+    // Define test architectures; an optional ROCm version argument is available
+    def nodes = new dockerNodes(nodeDetails, jobName, prj)
+
+    boolean formatCheck = false
+    boolean staticAnalysis = false
+
+    prj.timeout.test = 30
+    prj.defaults.ccache = false
+
+    def compileCommand = { platform, project -> runCompileCommand(platform, project, jobName, false) }
+
+    buildProject(prj, formatCheck, nodes.dockerArray, compileCommand, null, null, staticAnalysis)
+}
+
+def ci() {
+    String urlJobName = auxiliary.getTopJobName(env.BUILD_URL)
+
+    properties(auxiliary.addCommonProperties([pipelineTriggers([cron('0 2 * * 6')])]))
+
+    stage(urlJobName) {
+        runCI([ubuntu22:['gfx90a']], urlJobName)
+    }
+}
```

Comment:
What is this doing?

Reply:
This defines the execution environment for the pipeline. 

Removing the cron trigger.